vscode/src/extension.ts

```diff
@@ -514,17 +514,19 @@ export function activate(context: ExtensionContext): VSNetBeansAPI {
             vscode.window.showErrorMessage('Cannot find workspace path');
             return;
         }
+        
         const userDir = path.join(storagePath, "userdir");
         if (userDir && fs.existsSync(userDir)) {
-            const confirmation = await vscode.window.showInformationMessage('Are you sure you want to delete cache for this workspace?',
+            const confirmation = await vscode.window.showInformationMessage('Are you sure you want to delete cache for this workspace  and reload the window ?',
                 'Yes', 'Cancel');
             if (confirmation === 'Yes') {
-                await fs.promises.rmdir(userDir, {recursive : true});
-                const res = await vscode.window.showInformationMessage('Cache cleared successfully for this workspace', 'Reload window');
-                if (res === 'Reload window') {
-                    await vscode.commands.executeCommand('workbench.action.reloadWindow');
+                    stopClient(client).then(() => { deactivated=true;
```

Comment:
Since the parent function is already using `async/await`, can we refactor this callback-based code to use `async/await` for better consistency? 
This will also allow us to handle errors more effectively using `try/catch` blocks.

Reply:
Try to add catch and finally